File: 1744-Can-Eat/1744.go
func canEat(candiesCount []int, queries [][]int) []bool {
	n := len(candiesCount)

	// 前缀和
	sum := make([]int, n)
	sum[0] = candiesCount[0]
	for i := 1; i < n; i++ {
		sum[i] = sum[i-1] + candiesCount[i]
	}

	ans := make([]bool, len(queries))
	for i, q := range queries {
		favoriteType, favoriteDay, dailyCap := q[0], q[1], q[2]

		x1 := favoriteDay + 1
		y1 := (favoriteDay + 1) * dailyCap
		x2 := 1
		if favoriteType > 0 {
			x2 = sum[favoriteType-1] + 1
		}
		y2 := sum[favoriteType]

		ans[i] = !(x1 > y2 || y1 < x2)
	}
	return ans
}

File: 1744-Can-Eat/1744.py
class Solution:
  def canEat(self, candiesCount: List[int], queries: List[List[int]]) -> List[bool]:
    # 前缀和
    total = list(accumulate(candiesCount))

    ans = list()
    for favoriteType, favoriteDay, dailyCap in queries:
      x1 = favoriteDay + 1
      y1 = (favoriteDay + 1) * dailyCap
      x2 = 1 if favoriteType == 0 else total[favoriteType - 1] + 1
      y2 = total[favoriteType]

      ans.append(not (x1 > y2 or y1 < x2))
      
    return ans